telemetry/derive/tests/ui_fail/not_return_result.rs

```diff
@@ -0,0 +1,7 @@
+use iroha_core::wsv::{World, WorldStateView};
+use iroha_telemetry_derive::metrics;
+
+#[metrics(+"test_query", "another_test_query_without_timing")]
+fn execute(wsv: &WorldStateView<World>) -> iroha_core::Result {
```

Comment:
could we allow any type which can be converted to/from `std::result::Result`?

Reply:
Right now, any instance of such code is either a violation of our style guide, or a logic error.

Reply:
ok, idk, I had `eyre::Result` in mind

Reply:
Good example. How would you handle the status code for the response?

Reply:
I guess this is the contentious part your are wondering about:
```
if let Ok(_) = res {
    #successes
};
```

Reply:
I think the expanded macro would work for any type which has `Ok` as one of it's variants?

Reply:
Could you elaborate on this? I'm not sure I understand the problem